neurokit2/complexity/utils_complexity_coarsegraining.py

```diff
@@ -184,11 +186,16 @@ def complexity_coarsegraining(signal, scale=2, method="nonoverlapping", show=Fal
         # Relying on scipy is a fast alternative to:
         # pd.Series(signal).rolling(window=scale).mean().values[scale-1::]
         # https://stackoverflow.com/questions/13728392/moving-average-or-running-mean
-        coarse = scipy.ndimage.filters.uniform_filter1d(signal, size=scale, mode="nearest")
-        coarse = coarse[scale - 1 : :]
+        # coarse = scipy.ndimage.filters.uniform_filter1d(
+        #     signal, size=scale, mode="nearest"
+        # )
+        # coarse = coarse[scale - 1 : :]
+        coarse = complexity_embedding(signal, dimension=scale, delay=1).mean(axis=1)
```

Comment:
I updated the coarsegraining procedure but it doesn't seem to have entirely solved the issue:

![image](https://github.com/neuropsychology/NeuroKit/assets/8875533/6db7e714-96ce-4bb3-9886-97366f431bf4)


@hsyu001 let's just make sure that we have the same sample entropy results: would you mind computing SampEn with these parameters using your own algorithm:

```python
signal = [1, 2, 3, 5, 3, 1, 2, 4, 5, 7, 3, 2, 6, 2, 4, 8, 2]
tol = 2
```

With NK, this gives:
```python
nk.entropy_sample(signal, dimension=2, delay=3, tolerance=tol)
> (0.2831469172863898,
 {'Dimension': 2, 'Delay': 3, 'Tolerance': 2.0034572195207527})
```

Reply:
PS if you install NK from this branch using:

```
pip install https://github.com/neuropsychology/neurokit/zipball/fix_entropymultiscale --upgrade
```

 can you also make sure that the new coarsegraining gives the same results as expected?

Reply:
Hi Dr. Makowski :
The new coarse-graining function works fine for me, and the results are as expected as other methods. 
![image](https://github.com/neuropsychology/NeuroKit/assets/56075722/6b456d73-f6d4-409a-8962-a34221ed97b9)

Reply:
I have tried several tools that are commonly used to measure the sample entropy. Each tool has different results under different time delay parameter settings, I am not sure which one is correct.
![image](https://github.com/neuropsychology/NeuroKit/assets/56075722/0c1fb3ce-f9ef-4f4d-8c87-2d8878d6ba18)

Reply:
that's rather annoying... they might be some differences in the `tol` argument that gets postprocessed by some functions (e.g., adjusted under the hood for standard deviation?).

But in any case, could you help me isolate, from your modified_MMSE() code, the part that corresponds to the sampen computation: 

```python

def modified_MSE(data, scale, R = 0.15, m = 2):

    mmse_result = []
    
    r = R * np.std(data)
    
    for s in range(1, scale + 1):
        
        mean_data = []
        
        N = len(data)
               
        for i in np.arange(0, N - s + 1).reshape(-1):
                
            moving_mean = np.mean(data[np.arange(i, i + s)])
                
            mean_data.append(moving_mean)
        
        
        N = len(mean_data)
        
        Nn = 0
        
        Nd = 0
        
        delay = s
        
        
        for i in np.arange(0, N - (m + 1) * delay).reshape(-1):
            
            for j in np.arange(i + delay, N - m * delay).reshape(-1):
                
                if (np.abs(mean_data[i] - mean_data[j]) < r) and (np.abs(mean_data[i + delay] - mean_data[j + delay]) < r):
                    
                    Nn += 1
                    
                    if abs(mean_data[i + m * delay] - mean_data[j + m * delay]) < r:
                        
                        Nd += 1

        
        entropy = -np.log(Nd / Nn)
        
        mmse_result.append(entropy)
        
    return mmse_result
``` 

and put it in a separate function so I can run some tests with it, because yours seemed to be the one that gave some good results if we consider the pink noise curve as a benchmark

Reply:
Hi Dr. Makowski :
Ok,  I'll do that later...